### PR TITLE
Add settings to toggle attributes

### LINKED ISSUE #654: Custom Zeus attributes displays to hide certain attributes
**Is your enhancement related to a problem?**
Current overhauls of Zeus attributes displays will make Eden Editor Module 'MouduleCuratorSetAttributesObject_f'/'MouduleCuratorSetAttributesPlayer_f'/'MouduleCuratorSetAttributesGroup_f' all lose efficacy and the current attributes screen was getting crowded.

**Solution you'd like:**
A module in Eden Editor which similar to original Eden Editor Module like  'MouduleCuratorSetAttributesObject_f' to turn on or off certain attributes (example. the name blank of the unit) for certain Zeus unit during mission editting.

**Alternatives you've considered:**
A game setting like current ZEN faction filter.

**Additional context:**
No at the momment.

### PR DESCRIPTION
**Pull request by ampersand38**
**When merged this pull request will:**
- Some attributes take up a lot of screen space and are not often used by some users
- Provides settings to disable unwanted attributes
- Object and Group attributes are handled
- Close #654 
image

### CODE REVIEW

**Comment by mharis001** [comment]
I like the idea of having settings for enabling/disabling attributes but not sure about limiting it to only certain ones. A more generalized approach (like the faction filter) might be better but I can foresee some weird UX cases with this option for the "sub-attributes menus" and the buttons used to access them.

**Review by Kexanone — APPROVED** [review]

**Comment by ampersand38** [comment]
Ah, I was gonna get around to adding the settings to more attributes and see what mharis thought xD
Soon!

**Comment by ampersand38** [comment]
image
image

**Review by Kexanone — APPROVED** [review]
LGTM

File: addons/attributes/initAttributes.sqf [hunk 1]
```diff
@@ -518,7 +526,8 @@
             _x setWaypointType _type;
             if (_type == "SCRIPTED") then {_x setWaypointScript _script};
         } forEach SELECTED_WAYPOINTS;
-    }
+    },
+    {GVAR(enableWpName)}
```
Comment (mharis001) [inline addons/attributes/initAttributes.sqf:530]: Missing setting for this? Also, should be `enableWaypointType`.

File: addons/attributes/stringtable.xml [hunk 2]
```diff
@@ -368,5 +384,9 @@
             <Chinese>打開所選實體的Zeus屬性顯示。當選擇多個實體時（即由於選擇了一個小組），將使用懸停的實體。</Chinese>
             <Italian>Apre la visualizzazione degli attributi di Zeus per l'entità selezionata.\nQuando vengono selezionate più entità (ad es. a causa della selezione di un gruppo), verrà utilizzata l'entità al passaggio del mouse.</Italian>
         </Key>
+        <Key ID="STR_ZEN_Attributes_EnableAttribute_Description">
+            <English>Enables the related attribute(s) in the Zeus attributes display</English>
```
Comment (mharis001) [inline addons/attributes/stringtable.xml:388]: Missing period at the end. Also maybe don't need this generic description for all of the attribute settings.

**Review by mharis001 — CHANGES_REQUESTED** [review]

File: addons/attributes/initAttributes.sqf [hunk 3]
```diff
@@ -195,7 +195,7 @@
         [_buildings, _value] call EFUNC(building_markers,set);
     },
     {_entity getVariable [QEGVAR(building_markers,marker), ""] != ""},
-    {_entity isKindOf "Building"}
+    {GVAR(enableBuildingMarker) && _entity isKindOf "Building"}
```
Comment (mharis001) [inline addons/attributes/initAttributes.sqf:198]: Braces around `isKindOf` check.

File: addons/attributes/initAttributes.sqf [hunk 4]
```diff
@@ -348,7 +349,7 @@
         };
     },
     {""},
-    {IS_ADMIN_LOGGED || {!GETMVAR(ZEN_disableCodeExecution,false)}}
+    {GVAR(enableDebugConsole) && {IS_ADMIN_LOGGED || {!GETMVAR(ZEN_disableCodeExecution,false)}}}
```
Comment (mharis001) [inline addons/attributes/initAttributes.sqf:352]: I think this should be `enableExecute`.

File: addons/attributes/initAttributes.sqf (hunk 1 above)
Comment (ampersand38) [reply addons/attributes/initAttributes.sqf:530]: ah oops, yeah I was gonna stop at obj and grp

File: addons/attributes/stringtable.xml (hunk 2 above)
Comment (ampersand38) [reply addons/attributes/stringtable.xml:388]: Remove entirely, or just the first entry, or another alterative?

File: addons/attributes/initAttributes.sqf (hunk 1 above)
Comment (mharis001) [reply addons/attributes/initAttributes.sqf]: Why not include waypoints and markers as well, for completeness.

File: addons/attributes/stringtable.xml (hunk 2 above)
Comment (mharis001) [reply addons/attributes/stringtable.xml:388]: Remove entirely IMO.

**Comment by mharis001** [comment]
I added the remaining attributes/buttons including those in other components. A bit of a mess, but I guess there isn't really a better way.

**Review by mharis001 — APPROVED** [review]

**Pull request merged**